### PR TITLE
docs(guides): amend doInvoke and advanced API guide

### PR DESCRIPTION
**Pull request by snowypowers**
Update and rearrange some information in `doInvoke` and the advanced `doInvoke` guide

- Shifted the drop of gas workaround into basic `doInvoke`
- Renamed advanced `doInvoke` to be a more general API core functionality guide
- Rewrote the advanced guide to reflect more functionality.

### CODE REVIEW

**Comment by snowypowers** [comment]
@jeroenptrs I amended your guides. Please review.

File: docs/guides-adv_apicore.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+}
+```
+
+## Information Retrieval
+
+This is the first step of every method. Based on the properties `net` and `address` provided, `neon-js` will attempt to retrieve the required information in order to construct the transaction.
+
+`net` represents the network which `neon-js` will attempt to work on. Usual terms provided are `MainNet` or `TestNet` which are networks hard-coded within the SDK itself. If you are using a private network, you will need to add a new network in settings or provide the direct url of your 3rd party provider here (NeonDB or neoscan).
+
+`address` is the address of the account that you wish to send from. This is the address which the assets originate from. This field is used to query your balance or claims from the 3rd party provider.
```
Comment (jeroenptrs) [inline docs/guides-adv_apicore.md]: ... from which the assets ...

File: docs/guides-adv_apicore.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+For signing, the property `privateKey` from the configuration object is used to sign the object. An alternative to this is to provide a signing function under the property `signingFunction`.
+
+Instead of sending a user's private key to the config object, we can send the public key and a function that will sign the transaction.
+This function, the signingFunction, will receive the transaction and public key as parameters. Now, we can provide logic that retrieves the private key from our user - using the public key to do so - and signs the transaction when we retrieve the key.
+
+This the current way `neon-js` interacts with the Ledger Nano S for signing transactions.
+
+> Do note that the signing function has to return a Promise!
+
+```js
```
Comment (jeroenptrs) [inline docs/guides-adv_apicore.md]: Isn't it going to be confusing if we use ts and js interchangeably in the same doc?

Comment (snowypowers) [reply docs/guides-adv_apicore.md]: thats the only way i can present an interface. JS doesnt have this construct but honestly i would treat it more like pseudo code

**Comment by snowypowers** [comment]
@jeroenptrs typoes are corrected! Thanks for proofreading! Anything else you wanna add?

**Comment by jeroenptrs** [comment]
@snowypowers nah looks good! It's a good read!

**Pull request merged**